src/Data/Functor/Linear/Internal.hs

```diff
@@ -33,19 +33,21 @@ class Functor f where
 
 -- | Data 'Applicative'-s can be seen as containers which can be zipped
 -- together. A prime example of data 'Applicative' are vectors of known length
--- ('ZipLists' would be, if it were not for the fact that zipping them together
+-- ('ZipList's would be, if it were not for the fact that zipping them together
 -- drops values, which we are not allowed to do in a linear container).
 --
 -- In fact, an applicative functor is precisely a functor equipped with (pure
 -- and) @liftA2 :: (a ->. b ->. c) -> f a ->. f b ->. f c@. In the case where
 -- @f = []@, the signature of 'liftA2' would specialise to that of 'zipWith'.
 --
--- Intuitively, Data 'Applicative's can be seen as containers whose "number" of
--- elements is known at compile-time. This includes vectors of known length
--- but excludes 'Maybe', since this may contain either zero or one value.
--- Similarly, @((->) r)@ forms a Data 'Applicative', since this is a (possibly
--- infinitary) container indexed by @r@, while lists do not, since they may
--- contain any number of elements.
+-- Intuitively, the type of 'liftA2' means that 'Applicative's can be seen as
+-- containers whose "number" of elements is known at compile-time. This
+-- includes vectors of known length but excludes 'Maybe', since this may
+-- contain either zero or one value.  Similarly, @((->) r)@ forms a Data
+-- 'Applicative', since this is a (possibly infinitary) container indexed by
+-- @r@, while lists do not, since they may contain any number of elements.
+-- On the other hand, the type of 'pure' enforces that this "number" must be
+-- at least one, and so the container cannot be empty.
```

Comment:
Pure is an unrestricted function. I'm pretty sure `Const a` is a `Data.Applicative` (if `a` is a (linear) `Monoid`).

Reply:
Yup, my mistake. I was confusing "`f ()` is non-empty" with the container being non-empty.